cmd/minikube/cmd/start.go

```diff
@@ -1652,3 +1653,9 @@ func exitGuestProvision(err error) {
 	}
 	exit.Error(reason.GuestProvision, "error provisioning guest", err)
 }
+
+func validateNoAddonsForNoKubernetes() {
+	if viper.GetStringSlice(config.AddonListFlag) != nil && viper.GetBool(noKubernetes) {
```

Comment:
`viper.GetStringSlice(config.AddonListFlag)` isn't nil when no `addons` flag is passed, try running `minikube start --no-kubernetes` yourself.